compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1413,6 +1424,23 @@ object Parsers {
       }
     }
 
+    /** Replaces kind-projector's `*` in a list of types arguments with synthetic names,
+     *  returning the new argument list and the synthetic type definitions.
+     */
+    private def replaceKindProjectorPlaceholders(params: List[Tree]): (List[Tree], List[TypeDef]) = {
+      val tparams = new ListBuffer[TypeDef]
+
+      val newParams = params.map {
+        case Ident(tpnme.raw.STAR) =>
+          val name = tpnme.syntheticTypeParamName(tparams.length)
+          tparams += TypeDef(name, TypeBoundsTree(EmptyTree, EmptyTree))
```

Comment:
1. set the Param flag to make sure this is handled as a type parameter
2. set the position ("span") of the constructed tree to match the position of the user-written `*`, for better error messages and IDE support.

```suggestion
          tparams += TypeDef(name, TypeBoundsTree(EmptyTree, EmptyTree))
            .withFlags(Param).withSpan(param.span)
```
There's another use of TypeDef below which should do something similar (or even better, be refactored so there's only one point where TypeDef is used)

Reply:
I don't see a way to use the `*` span without ending up with a "child positions in wrong order" error:
```scala
java.lang.AssertionError: assertion failed: position error, child positions overlap or in wrong order
parent         = [_$5] =>> Either[Int, _$5]
1st child      = _$5
1st child span = [117..118]
2nd child      = Either[Int, _$5]
2nd child span = <105..115> while compiling tests/pos-special/kind-projector.scala
Fatal compiler crash when compiling: tests/pos-special/kind-projector.scala:
assertion failed: position error, child positions overlap or in wrong order
parent         = [_$5] =>> Either[Int, _$5]
1st child      = _$5
1st child span = [117..118]
2nd child      = Either[Int, _$5]
2nd child span = <105..115>
dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
dotty.tools.dotc.ast.Positioned.check$1(Positioned.scala:183)
```
It seems like in this case it just might not be possible to point to the user-written `*`?